src/main/java/sirius/biz/jobs/batch/file/FileImportJob.java

```diff
@@ -287,9 +300,17 @@ private String getTargetPath(ExtractedFile extractedFile) {
         String targetPath = extractedFile.getFilePath();
         if (flattenAuxiliaryFileDirs) {
             return Files.getFilenameAndExtension(targetPath);
-        } else {
-            return targetPath;
         }
+
+        // Drops the parent directory from the path if we match it with the beginning:
+        // parent -> /foo/bar
+        // path obtained from archive -> /foo/bar/zoo/file
+        // result -> /zoo/file
+        if (targetPath.toLowerCase().startsWith(auxFilesParentDirectory.toLowerCase())) {
+            return targetPath.substring(auxFilesParentDirectory.length());
+        }
```

Comment:
Not sure whether I get the purpose of these lines ... if files in the archive already contain the "prefix", it will be dropped, causing such files to be moved up (relatively)?

If this is the case, the unpacked file structure will be different from the packed one? Wouldn't that be too much magic?

Reply:
Is a _desired magic_ which tries to cover a common user failure: the user wants its files to be unpacked in a parent folder, but the files inside the archive are already structured under the folder. By not doing it and simply exploding the archive in the target would yield `${root}/foo/bar/foo/bar/file3` in the example of the PR description.

Reply:
You could add to the comment above the code that the removed base path is later added again